languages/java/src/main/java/de/jplag/java/TokenGeneratingTreeScanner.java

```diff
@@ -391,7 +401,7 @@ public Object visitExports(ExportsTree node, Object p) {
 
     @Override
     public Object visitErroneous(ErroneousTree node, Object p) {
-        parser.increaseErrors();
+        parsingException = new ParsingException(file, "error while visiting %s".formatted(node));
```

Comment:
Can this be called multiple times? If yes, we may accidentally override this exception.

Reply:
I did not check the documentation in detail but I guess it was possible. As previously we only shared the count of errors, I didn't care to overwrite error metadata but I can adapt it to support logging multiple exceptions.